src/include/sof/audio/audio_stream.h

```diff
@@ -202,6 +209,30 @@ static inline uint32_t audio_stream_sample_bytes(const struct audio_stream *buf)
 	return get_sample_bytes(buf->frame_fmt);
 }
 
+/**
+ * Set frame_align_shift and frame_align of stream according to byte_align and
+ * frame_align_req alignment requirement, and these two feature will be used in
+ * audio_stream_get_avail_frames_aligned and audio_stream_get_free_frames_aligned
+ * to calculate the available frames.
```

Comment:
This initialization function seems critical. Please explain here why these values are constant and WHEN they can be initialized whereas the rest is variable and  computed repeatedly (WHEN / how often exactly?). Please take the perspective of the developer who will apparently NOT call `audio_stream_get_avail_frames_aligned()` directly.

Also, what happens when someone forgets to call it?